programs/move_loader/src/processor.rs

```diff
@@ -36,38 +35,44 @@ use vm_runtime::{
 };
 use vm_runtime_types::value::Value;
 
-pub fn process_instruction(
-    _program_id: &Pubkey,
-    keyed_accounts: &mut [KeyedAccount],
-    data: &[u8],
-) -> Result<(), InstructionError> {
-    solana_logger::setup();
+/// Instruction data passed to perform an loader operation, must be based
+/// on solana_sdk::loader_instruction::LoaderInstruction
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub enum MoveLoaderInstruction {
```

Comment:
This PR decouples the runtime from enforcing a contract on a program or loader.  But, I'd like to keep a common `LoaderInstruction` where it makes sense.  Ideally, it would be nice to be able to extend/inherit from `LoaderInstruction`.  @garious any suggestions on how we might do this more cleanly?

Reply:
I don't know of a way. I'm okay with the duplication. It's not much different than how each program has an InitializeAccount instruction.